include/exadg/grid/grid.h

```diff
@@ -80,30 +126,69 @@ class Grid
   create_triangulation(
     GridData const &                                          data,
     std::function<void(dealii::Triangulation<dim> &)> const & create_coarse_triangulation,
+    unsigned int const                                        global_refinements,
     std::vector<unsigned int> const & vector_local_refinements = std::vector<unsigned int>())
   {
     do_create_triangulation(data,
                             create_coarse_triangulation,
-                            true /* do refine */,
+                            global_refinements,
                             vector_local_refinements);
+
+    if(data.create_coarse_triangulations)
+    {
+      // in case of a serial or distributed triangulation, deal.II can automatically generate the
+      // coarse grid triangulations
+      if(data.triangulation_type == TriangulationType::Serial or
+         data.triangulation_type == TriangulationType::Distributed)
+      {
+        coarse_triangulations =
+          dealii::MGTransferGlobalCoarseningTools::create_geometric_coarsening_sequence(
+            *triangulation,
+            BalancedGranularityPartitionPolicy<dim>(
+              dealii::Utilities::MPI::n_mpi_processes(triangulation->get_communicator())));
+      }
+      else if(data.triangulation_type ==
+              TriangulationType::FullyDistributed) // we need to generate the coarse triangulations
+                                                   // explicitly
+      {
+        // TODO
+        // construct all the coarser triangulations
+      }
```

Comment:
I don't think this is the correct logic. To me it looks like, you are creating the active mesh first and based on that you are creating the coarser one. The logic should be c`reate coarse mesh`-`copy it`-`refine mesh`-`copy it`-....